stan/math/prim/mat/fun/opencl_copy.hpp

```diff
@@ -0,0 +1,132 @@
+#ifndef STAN_MATH_PRIM_MAT_FUN_OPENCL_COPY_HPP
+#define STAN_MATH_PRIM_MAT_FUN_OPENCL_COPY_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/gpu/opencl_context.hpp>
+#include <stan/math/gpu/matrix_gpu.hpp>
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <stan/math/prim/scal/err/check_size_match.hpp>
+#include <CL/cl.hpp>
+#include <iostream>
+#include <vector>
+#include <algorithm>
+
+namespace stan {
+namespace math {
+
+/**
+ * Copies the source Eigen matrix to
+ * the destination matrix that is stored
+ * on the GPU.
+ *
+ * @tparam T type of data in the Eigen matrix
+ * @param src source Eigen matrix
```

Comment:
Minor: these should match the order of the arguments. I'll try to fix it on your branch directly.